server/build-plugins.js

```diff
@@ -33,6 +35,47 @@ const getPackageName = name => {
   }
 };
 
+async function installRemotePackages(installPackages) {
+  // check if connected to internet
+  // if not, skip npm install
+  const EXTERNAL_URI = process.env.EXTERNAL_URI || 'npmjs.com';
+  await require('dns').lookup(EXTERNAL_URI, async err => {
+    if (err && err.code === 'ENOTFOUND')
+      logger.error(`Can't reach npm servers. Skipping npm install`);
+    else {
+      logger.info('Installing plugin packages...');
+      await execSync(
+        `npm install --no-save ${installPackages.join(' ')} --production`,
+        {
+          stdio: [0, 1, 2],
+          cwd: resolve(__dirname, '..')
+        }
+      );
+      logger.info('Done installing plugins');
+    }
+  });
+}
+
+function symlinkLocal(packageName) {
+  let name = packageName;
+  // set temporary name for symlink since fs can't overwrite
+  // with symlink method
+  if (fs.existsSync(resolve(modulesDirectory, packageName)))
+    name = `_${packageName}`;
+
+  fs.symlinkSync(
+    resolve(homePrefix, 'local_plugins', packageName),
+    resolve(modulesDirectory, name)
+  );
+
+  // now rename in order to overwrite
```

Comment:
Maybe we should maintain a backup of the module that we are overwriting? I feel like its common for software to do that when its manipulating the filesystem

Reply:
I did feel a little bad about deleting what's there. My feeling though is that anything in node_modules basically has a backup via npm (or whatever registry it was installed from). I feel like renaming would just add clutter since there's no real reference saved to it (and what if there's already a backup? should it overwrite that, or save a separate reference?).

Reply:
Yeah that is true, maybe a `logger.info` for each package that is overwritten? So that users won't run into a situation where they think they have a specific version of a package but they don't because it was silently replaced

Reply:
Yeah, good idea.